Add readme and template for ADRs

palemtnrider opened the pull request:
<!-- Describe what has changed in this PR -->
**What changed?**
Adding a readme describing ADRs and a template

<!-- Tell your future self why have you made these changes -->
**Why?**
We need to start capturing architectural decisions and this is a lightweight which is commonly used.

<!-- How have you verified this change? Tested locally? Added unit test/integration/acceptance test(s)? -->
**How did you test it?**
N/A
<!-- Is it notable for release? e.g. schema updates, configuration or data migration required? If so, please mention it. -->
**Release notes**
N/A

<!-- Are there any documentation updates that should be made for these changes? -->
**Documentation Changes**
N/A

bigkevmcd reviewed (commented):
Can I propose using https://github.com/npryce/adr-tools instead?

This reduces the cost of creating and maintaining the ADRs.

palemtnrider commented:
Let me give adr-tools a try.  I don't mind filling out a template but if that makes it even easier to capture the info I'm all for it.

palemtnrider commented:
@bigkevmcd I used adr-tools to generate #458 and it does simplify the process a bit.  I've updated the readme to reference adr-tools and updated the template to match the adr-tools one for those that don't want to install adr-tools
